src/rnp/rnp.c

```diff
@@ -496,12 +493,31 @@ setoption(rnp_cfg_t *cfg, int *cmd, int val, char *arg)
         *cmd = val;
         break;
     case CMD_DEARMOR:
-    case CMD_ENARMOR:
         *cmd = val;
-        rnp_cfg_setint(
-          cfg, CFG_ARMOR_DATA_TYPE, armor_str_to_data_type(arg, arg ? strlen(arg) : 0));
+        break;
+    case CMD_ENARMOR: {
+        pgp_armored_msg_t msgt = PGP_ARMORED_UNKNOWN;
+        *cmd = val;
+
+        if (arg) {
+            if (!strncmp("msg", arg, strlen(arg))) {
```

Comment:
I don't think there is a reason to duplicate this code (nevertheless, I think I now understand your point).
So,
```
struct {
 uint8_t ID,
 char *msg_data_type,
 char *cli_data_type,
} name[] = {
  { PGP_ARMORED_PUBLIC_KEY, "pubkey", "BEGIN PGP PUBLIC KEY" },
  .....
};
```
and then ```armor_str_to_data_type``` gets additional argument and does ```for``` inside to find correct id. That's how we do do in some other places

Reply:
I don't like this because 'BEGIN PGP PUBLIC KEY' is the constant from data layer which implements RFC, which does not depend on user in any way. While 'pubkey' is a CLI parameter which should be present on the CLI layer only. And mixing those layers in code will bring us back to GnuPG :)

Reply:
Nop, that's not a problem, because you can add an argument to ``armor_str_to_data_type`` which will indicate what too look for.

Reply:
Why that's not a problem? This is a completely wrong approach when entities from different parts of the library are mixed together just to save few lines of code. And, even now, command line argument does not directly fit to `armor_str_to_data_type` since it must not allow cleartext message.

@dewyatt , @ronaldtse what do you think? This is actually minor issue but we should follow some architectural logic and separate entities based on their usage.

Reply:
Ok, that's minor issue indeed. It seems it's faster if I'll fix it later, then explain.

Reply:
@ni4 I agree we should isolate the architecture first and then attempt to DRY the code. Maybe there are ways we can share those functions without polluting the architecture.